pandas/core/internals/managers.py

```diff
@@ -411,7 +411,10 @@ def apply(self, f: str, filter=None, **kwargs):
                     axis = obj._info_axis_number
                     kwargs[k] = obj.reindex(b_items, axis=axis, copy=align_copy)
 
-            applied = getattr(b, f)(**kwargs)
+            if callable(f):
```

Comment:
is this strictly necessary? meaning happy to require only callables here (would require some changing)

Reply:
all of our existing usages pass strings here to get at Block methods.  i think @WillAyd had a suggestion about re-working Block.apply to do str vs callable handling there; that should be its own PR

Reply:
k, yeah this whole section could use some TLC